### ISSUE TITLE
Authenticated section with Markdown Pages?

### ISSUE THREAD

**Issue by vinayan3** [gatsbyjs/gatsby#9900]
I'm trying to have a set of markdown pages behind a login wall. Is there a good example of this? I have tried to use ` gatsby-plugin-create-client-paths` and a  `Router`  at the base path of the auth area which I have as `/app/`. In the function `createPages` in ` gatsby-node.js`  I put the path of the markdown files as `/app/{post.node.fields.slug}`.  After that how would I get the router to have the routes to the Markdown pages?   If there was a way to pull in the Component for each of the pages then everything would be solved.

Any guidance or examples would be appreciated.

Update:
I've been able to dynamically generate the Components  for each Markdown file instead of creating pages in `gatsby-node.createPages`. Is this the best way? Is there a more idiomatic way to do this?

**Comment by justinformentin** [comment]
First, have you read through the docs? The [official tutorial](https://www.gatsbyjs.org/tutorial/) should answer most of your question, it has a ton of info. For auth, check out the [entry in the docs about it](https://www.gatsbyjs.org/docs/building-a-site-with-authentication/)

It has links to steps to take, examples, and tutorials. This is a good starting place. If you want some more content to wrap your head around conditionally rendered routes, take a look at these auth0 and firebase specific example repos. 

[Auth0 react examples](https://github.com/auth0-samples/auth0-react-samples/tree/master/01-Login)

[Gatsby Firebase Example](https://github.com/taming-the-state-in-react/gatsby-firebase-authentication)

And to just answer your question about creating pages specifically, yes. Files in `/src/pages/` should really just be singular, unchanging, stands-on-their-own pages. Like an About page, a Home page, a Blog page, a Contact page, etc. And then pages that are created programmatically, generated from markdown files through gatsby-node.js should be things like blog posts, or other items that there are a number of identical type. 

Imagine how much of a pain it would be to create 50 individual components if you have a large blog. Instead, you just make 50 markdown files that get generated into post pages, and then a single `/src/templates/post.jsx`.

**Comment by vinayan3** [comment]
Yes I did go through the official tutorial and the authenticated sections.

I have settled on using conditional routes for the Markdown pages I need behind the authentication. I won't be manually creating each of those markdown pages but instead using a GraphQL query to get the MarkDown content and passing it into a template. 

I was hoping there would be a way for me to get access to the Global router that Gatsby may  or may not be creating and adjusting how it does routing.

I'm closing the issue because I won't need any further assistance. I really appreciate your thoughtful response.
